boa/__init__.py

```diff
@@ -45,17 +45,41 @@ def set_env(new_env):
     Env._singleton = new_env
 
 
+# Simple context manager which functions like the `open()` builtin -
+# if simply called, it never calls __exit__, but if used as a context manager,
+# it calls __exit__ at scope exit
+class _TmpEnv:
+    def __init__(self, new_env):
+        global env
```

Comment:
why do we need the global env and Env._singleton?

Reply:
global env is how the user accesses -- `boa.env`.

`Env._singleton` is arguably redundant, but i think it's good practice as a matter of hygiene to have singletons scoped to their classes.